logger.js

```diff
@@ -64,10 +72,21 @@ function pinoLogger (opts, stream) {
     req.id = genReqId(req)
     req.log = res.log = logger.child({req: req})
     res[startTime] = res[startTime] || Date.now()
-    if (!req.res) { req.res = res }
 
     if (autoLogging) {
-      res.on('finish', onResFinished)
+      // calculate here whether to log success, but apply this flag up to the last minute.
+      // this is to ensure that the event listener still executes (as there may be other logic that is needed).
+      var shouldLogSuccess = true
```

Comment:
Can you move this to the top? It's weird to have a variable definition within an if that it's used outside of scope.

Reply:
even if autologging = false? that shouldn't even execute, that is why it is inside the IF.

Reply:
I mean, just move `var autologging` at the top of the function.

Reply:
fixed